src/service/resources-impl.js

```diff
@@ -214,6 +214,20 @@ export class Resources {
     return this.resources_.slice(0);
   }
 
+  /**
+   * Returns a subset of resources which is identified as being in the current
+   * viewport.
+   * @return {!Array<!Resource>}
+   */
+  getResourcesInViewport() {
+    // TODO: ask @dvoytenko if we need to measure here.
+    const viewportRect = this.viewport_.getRect();
+    return this.get().filter(r => {
```

Comment:
Don't do `get`, just access `this.resources_` directly. `get` clones this list which you don't really need here.